dir.rb

```diff
@@ -0,0 +1,34 @@
+class Dir
+  #
+  #  call-seq:
+  #     Dir.new( string ) -> aDir
+  #     Dir.new( string, encoding: enc ) -> aDir
+  #
+  #  Returns a new directory object for the named directory.
+  #
+  #  The optional <i>encoding</i> keyword argument specifies the encoding of the directory.
+  #  If not specified, the filesystem encoding is used.
+  #
+  def initialize(path, encoding: Encoding.find("filesystem"))

```

Comment:
As `fsenc` is initialized in `dir_initialize`, the default value of `encoding` seems unnecessary to be filesystem encoding.